rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release/Sw360ReleaseService.java

```diff
@@ -1361,4 +1377,59 @@ public void unsubscribeRelease(User user, String releaseId) throws TException {
         ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
         sw360ComponentClient.unsubscribeRelease(releaseId, user);
     }
+
+    public List<Map<String,String>> getReleaseLicenseInfo(Release rel, User sw360User, String attachContentId) throws TException{
+        ThriftClients thriftClients = new ThriftClients();
+        LicenseInfoService.Iface licenseClient = thriftClients.makeLicenseInfoClient();
+        List<LicenseInfoParsingResult> licenseInfos = licenseClient.getLicenseInfoForAttachment(rel, attachContentId, false, sw360User);
+        Map<String, LicenseNameWithText> licenseStore = Maps.newHashMap();
+        List<Map<String, String>> licenses = Lists.newArrayList();
+        licenseInfos.forEach(licenseInfoResult ->
+                addLicenseInfoResultToJsonSerializableLicensesList(licenseInfoResult, licenses, licenseStore::put));
+        return licenses;
+    }
+
+    private void addLicenseInfoResultToJsonSerializableLicensesList(LicenseInfoParsingResult licenseInfoResult,
+                                                                    List<Map<String, String>> licenses,
+                                                                    BiConsumer<String, LicenseNameWithText> storeLicense) {
+        switch (licenseInfoResult.getStatus()) {
+            case SUCCESS:
+                Set<LicenseNameWithText> licenseNamesWithTexts = nullToEmptySet(licenseInfoResult.getLicenseInfo().getLicenseNamesWithTexts());
+                List<Map<String, String>> licensesAsObject = licenseNamesWithTexts.stream()
+                        .filter(licenseNameWithText -> !Strings.isNullOrEmpty(licenseNameWithText.getLicenseName())
+                                || !Strings.isNullOrEmpty(licenseNameWithText.getLicenseText())).map(licenseNameWithText -> {
+                            // Since the license has no good identifier, we create one and store the license
+                            // in the session. If the final report is generated, we use the identifier to
+                            // identify the licenses to be excluded
+                            // FIXME: this could be changed if we scan the attachments once after uploading
+                            // and store them as own entity
+                            String key = UUID.randomUUID().toString();
```

Comment:
Also, if the key is derived from the `licenseNameWithText` object instead of generating a random UUID, license store can be used more efficiently.